Common/include/option_structure.inl

```diff
@@ -70,220 +70,119 @@ public:
+template<typename Scalar>
+class COptionScalar : public COptionBase {
+protected:
+  Scalar& field; // Reference to the fieldname
+  const Scalar def; // Default value
+  const string name; // identifier for the option
+  const string typeName; // name for the scalar type
 
 public:
-  COptionDouble(string option_field_name, su2double & option_field, su2double default_value) : field(option_field) {
-    this->def = default_value;
-    this->name = option_field_name;
+  COptionScalar() = delete;
+
+  COptionScalar(const string& type_name,
+                const string& option_field_name,
+                Scalar& option_field,
+                Scalar default_value) :
+    field(option_field),
+    def(default_value),
+    name(option_field_name),
+    typeName(type_name) {
   }
 
-  ~COptionDouble() override {};
+  ~COptionScalar() = default;
+
   string SetValue(vector<string> option_value) override {
     COptionBase::SetValue(option_value);
-    // check if there is more than one value
-    string out = optionCheckMultipleValues(option_value, "su2double", this->name);
-    if (out.compare("") != 0) {
-      return out;
-    }
-    istringstream is(option_value[0]);
-    su2double val;
-    if (is >> val) {
-      this->field = val;
-      return "";
-    }
-    return badValue(option_value, "su2double", this->name);
-  }
-  void SetDefault() override {
-    this->field = this->def;
-  }
-};
 
-class COptionString : public COptionBase {
-  string & field; // Reference to the fieldname
-  string def; // Default value
-  string name; // identifier for the option
+    string out = optionCheckMultipleValues(option_value, typeName, name);
+    if (!out.empty()) return out;
 
-public:
-  COptionString(string option_field_name, string & option_field, string default_value) : field(option_field) {
-    this->def = default_value;
-    this->name = option_field_name;
-  }
+    istringstream is(option_value.front());
+    if (is >> field) return "";
 
-  ~COptionString() override {};
-  string SetValue(vector<string> option_value) override {
-    COptionBase::SetValue(option_value);
-    // check if there is more than one value
-    string out = optionCheckMultipleValues(option_value, "su2double", this->name);
-    if (out.compare("") != 0) {
-      return out;
-    }
-    this->field.assign(option_value[0]);
-    return "";
+    return badValue(option_value, typeName, name);
   }
+
   void SetDefault() override {
-    this->field = this->def;
+    field = def;
   }
 };
 
-class COptionInt : public COptionBase {
-  int & field; // Reference to the feildname
-  int def; // Default value
-  string name; // identifier for the option
-
+class COptionDouble : public COptionScalar<su2double> {
 public:
-  COptionInt(string option_field_name, int & option_field, int default_value) : field(option_field) {
-    this->def = default_value;
-    this->name = option_field_name;
+  template<class... Ts>
+  COptionDouble(Ts&&... args) :
```

Comment:
Yes this only works at compile time, the variadics are just to avoid repeating the constructor arguments.
The rvalue ref is to allow this to work with literal values, using just Ts& it would not be possible to bind a literal (or any rvalue) because it does not have an address, in those cases Ts&& collapses to just Ts.